civisocial.php

```diff
@@ -196,6 +206,16 @@ function civisocial_civicrm_postProcess($formName, &$form) {
       CRM_Civisocial_BAO_FacebookEvent::create($params);
     }
   }
+  elseif ('CRM_Event_Form_Registration_Confirm' == $formName) {
+    // Check if user chose to RSVP on Facebook
+    if (isset($form->_submitValues['facebook_rsvp_event'])) {
+      $eventId = $form->get('id');
+      $qfKey = $form->get('qfKey');
+      $thankyouUrl = rawurlencode(CRM_Utils_System::url("civicrm/event/register?_qf_ThankYou_display=true&qfKey={$qfKey}", NULL, TRUE));
+      $redirectUrl = CRM_Utils_System::url("civicrm/civisocial/event/rsvpfacebookevent?event_id={$eventId}&thankyou_url={$thankyouUrl}", NULL, TRUE);
```

Comment:
I didn't think about this. I think redirecting from Thank You pages' preProcess and coming back to the page after RSVPing should solve the problem. I will try that.